lib/import/brca/providers/salisbury/salisbury_handler.rb

```diff
@@ -13,56 +13,223 @@ def process_fields(record)
+              process_status_genes(status_genes, 10, genotype, genotypes, raw_record)
+            elsif NEGATIVE_STATUS.include? @status
+              process_status_genes(status_genes, 1, genotype, genotypes, raw_record)
+            elsif POSITIVE_STATUS.include?(@status) || @status.match(/^variant*/ix)
+              process_status_genes(status_genes, 2, genotype, genotypes, raw_record)
+            end
+          end
+
+          def prepare_gene_status_hash(record)
+            record.raw_fields.each do |raw_record|
+              assign_status_var(raw_record)
+
+              status_genes = extract_genes(%w[test genotype], raw_record)
+
+              if @status_genes_hash[status_genes]
+                @status_genes_hash[status_genes] << @status
+              else
+                @status_genes_hash[status_genes] = [@status]
+              end
+            end
+          end
+
+          def extract_genes(fields, raw_record)
+            status_genes = []
+            fields.each do |field|
+              result = raw_record[field]&.scan(BRCA_REGEX)&.flatten&.uniq
+              if result.present?
+                status_genes = result
+                break
+              end
+            end
+            status_genes
+          end
+
+          def assign_status_var(raw_record)
+            @status = raw_record['status']&.downcase
+          end
+
+          # Use priority if more than one status is present for same gene for a given record
+          def process_multi_status_genes(status_genes, status_found, genotype, genotypes, raw_record)
+            if (status_found & POSITIVE_STATUS) || status_found.map { |e| e.match(/^variant/) }
+              process_status_genes(status_genes, 2, genotype, genotypes, raw_record)
+            elsif status_found & ABNORMAL_STATUS
+              process_status_genes(status_genes, 10, genotype, genotypes, raw_record)
+            elsif status_found & NEGATIVE_STATUS
+              process_status_genes(status_genes, 1, genotype, genotypes, raw_record)
+            elsif status_found & FAILED_TEST
+              process_status_genes(@all_genes, 9, genotype, genotypes, raw_record)
+            elsif status_found & UNKNOWN_STATUS
+              process_status_genes(status_genes, 4, genotype, genotypes, raw_record)
+            end
+          end
+
+          def process_status_genes(genes, status, genotype, genotypes, record)
+            return unless genes&.all? { |gene| @all_genes.include?(gene) }
+
+            genes.each do |gene|
+              @all_genes -= [gene]
+              genotype_new = genotype.dup
+              genotype_new.add_gene(gene)
+              genotype_new.add_status(status)
+              if [2, 10].include? status
+                assign_variantpathclass_record(genotype_new)
+                variant = record['genotype']
+                process_variants(genotype_new, variant) if positive_record?(genotype_new) && variant.present?
```

Comment:
How do you know you are matching the variant with the correct gene here?

Reply:
Its now fixed in latest commit.